**SM339 &#x25aa; Applied Statistics &#x25aa; Spring 2023 &#x25aa; Uhan**

# Lesson 30. Multiple Logistic Regression &ndash; Part 2 

In this lesson, we will continue to use the `MedGPA` data, which records whether 55 medical school applicants were accepted.

In [1]:
library(Stat2Data)
data(MedGPA)
head(MedGPA)

Accept Acceptance Sex BCPM GPA  VR PS WS BS MCAT Apps
1 D      0          F   3.59 3.62 11  9 9   9 38    5  
2 A      1          M   3.75 3.84 12 13 8  12 45    3  
3 A      1          F   3.24 3.23  9 10 5   9 33   19  
4 A      1          F   3.74 3.69 12 11 7  10 40    5  
5 A      1          F   3.53 3.38  9 11 4  11 35   11  
6 A      1          M   3.59 3.72 10  9 7  10 36    5

We will focus on the following variables:

| Variable | Description |
| :- | :- | 
| `GPA` | Applicant's college grade point average |
| `MCAT` | Applicant's MCAT (Medical College Admission Test) score |
| `Sex` | F for female, M for male |
| `Acceptance` | 1 if accepted, 0 if not accepted |

Unless otherwise stated, use a significance level of $\alpha = 0.05$.
Throughout this lesson, let $\pi = P(\mathit{Acceptance} = 1)$.

## Example 1

__Does probability of acceptance differ by sex?__

### a.

Fit a model that uses sex to predict acceptance.

In [2]:
# Solution
fit <- glm(Acceptance ~ Sex, data = MedGPA, family = binomial)
summary(fit)


Call:
glm(formula = Acceptance ~ Sex, family = binomial, data = MedGPA)

Deviance Residuals: 
   Min      1Q  Median      3Q     Max  
-1.435  -1.084   0.940   0.940   1.274  

Coefficients:
            Estimate Std. Error z value Pr(>|z|)
(Intercept)   0.5878     0.3944   1.490    0.136
SexM         -0.8109     0.5528  -1.467    0.142

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 75.791  on 54  degrees of freedom
Residual deviance: 73.594  on 53  degrees of freedom
AIC: 77.594

Number of Fisher Scoring iterations: 4


Note that `Sex` is a categorical variable with two "levels" (categories): 

In [3]:
levels(MedGPA$Sex)

[1] "F" "M"

For categorical variables, R automatically creates a binary variable for every level except the first:

$$ \mathit{SexM} = \begin{cases}
1 & \text{if $\mathit{Sex}$ = M}\\
0 & \text{otherwise}
\end{cases}$$

The first level (in this case, 'F') is used as the reference category.

You should find the fitted model is

$$ \text{logit}(\hat{\pi}) = 0.5878 - 0.8109 \mathit{SexM} $$

### b.

Interpret the estimated slope of $\mathit{SexM}$, in terms of an odds ratio.

In [4]:
# Solution
exp(-0.81)

[1] 0.4448581

*Write your answer here. Double-click to edit.*

*Solution.*
The odds of acceptance for males are $e^{-0.81} = 0.44$ times as high as for females.

### c.

Provide a 95\% confidence interval for the true odds ratio.

In [5]:
# Solution
alpha <- 0.05
z <- qnorm(1 - alpha / 2, mean = 0, sd = 1)

slope.ci.lower <- -0.8109 - z * 0.5528
slope.ci.upper <- -0.8109 + z * 0.5528

or.ci.lower <- exp(slope.ci.lower)
or.ci.upper <- exp(slope.ci.upper)

or.ci.lower
or.ci.upper

[1] 0.1504134

[1] 1.313333

### d.

Is there significant evidence that the probability of acceptance differs by sex? Justify your answer.

*Write your answer here. Double-click to edit.*

*Solution*.
The $z$-test for the slope of $\mathit{SexM}$ results in a $p$-value of 0.142, which is greater than $\alpha$.
Therefore, we fail to reject the null hypothesis.
We conclude that we do **not** have significant evidence that the probability of acceptance differs by sex. 

## Example 2

__After accounting for GPA, does probability of acceptance differ by sex?__

### a.

Fit a model that uses GPA and sex to predict acceptance. 

In [6]:
# Solution
fit <- glm(Acceptance ~ GPA + Sex, data = MedGPA, family = binomial)
summary(fit)


Call:
glm(formula = Acceptance ~ GPA + Sex, family = binomial, data = MedGPA)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-1.8171  -0.9753   0.2978   0.7194   2.4623  

Coefficients:
            Estimate Std. Error z value Pr(>|z|)    
(Intercept) -21.0680     6.4025  -3.291 0.001000 ***
GPA           6.1324     1.8283   3.354 0.000796 ***
SexM         -1.1697     0.7178  -1.629 0.103210    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 75.791  on 54  degrees of freedom
Residual deviance: 53.945  on 52  degrees of freedom
AIC: 59.945

Number of Fisher Scoring iterations: 5


You should find the fitted model is

$$ \text{logit}(\hat{\pi}) = -21.0680 + 6.1324 \mathit{GPA} - 1.1697 \mathit{SexM} $$

Therefore, for males, the model is
\begin{align*}
\text{logit}(\hat{\pi}) & = -21.0680 + 6.1324 \mathit{GPA} - 1.1697 (1)\\
& = -22.2377 + 6.1324 \mathit{GPA}
\end{align*}
and for females,
\begin{align*}
\text{logit}(\hat{\pi}) & = -21.0680 + 6.1324 \mathit{GPA} - 1.1697 (0)\\
& = -21.0680 + 6.1324 \mathit{GPA}
\end{align*}

### b.

Sketch the plots of the fitted model (in probability form) for males and females.
Use the axes provided at the end of the notes for Part 1.

*Solution.* The plots for both models will have the same curvature, since they both have the same slope on GPA.

The vertical midpoint for the male model occurs at $\mathit{GPA} = 22.2377 / 6.1324 = 3.6263$, and the vertical midpoint for the female model occurs at $\mathit{GPA} = 21.0680 / 6.1324 = 3.4356$.

### c.

Is the model useful overall? Conduct an appropriate test to decide.

In [7]:
# Solution
G <- 75.791 - 53.945
p.value <- 1 - pchisq(G, df = 2)

G
p.value

[1] 21.846

[1] 1.803854e-05

*Write your answer here. Double-click to edit.*

*Solution.* 

LRT for model utility:

1. $H_0: \beta_1 = \beta_2 = 0$ versus $H_A: \text{at least one } \beta_i \ne 0$
2. $G = 75.791 - 53.945 = 21.846$
3. $p$-value = $1.804 \times 10^{-5}$
4. Reject $H_0$, because $p$-value is less than $\alpha = 0.05$. We see significant evidence that the model is useful.

### d.

Answer the motivating question for this example. Justify your answer.

*Write your answer here. Double-click to edit.*

*Solution.* We fail to reject the null hypothesis of a $z$-test for the coefficient of $\mathit{SexM}$, because the $p$-value = 0.103210 > 0.05 = $\alpha$. 

No, we do not see significant evidence that the probability of acceptance differs by sex.

### e.

Estimate the odds ratio of acceptance for males with a 3.5 GPA versus females with a 3.5 GPA.

In [8]:
# Solution
odds.male <- exp(-21.0680 + 6.1324*3.5 - 1.1697*1)
odds.female <- exp(-21.0680 + 6.1324*3.5 - 1.1697*0)

odds.male / odds.female

[1] 0.3104601

Your answer should be $e^{\hat{\beta}_2} = e^{-1.1697}$:

In [9]:
exp(-1.1697)

[1] 0.3104601

### f.

Interpret the coefficient of $\mathit{SexM}$ in terms of an odds ratio. 

*Write your answer here. Double-click to edit.*

*Solution.* For any fixed GPA, the odds of a male being accepted are 0.31 times as high as for a female.

Caveat: in part d, we showed that we actually don't have significant evidence that $\beta_2 \ne 0$. 

### g.

Estimate the odds ratio of acceptance for males with a 3.8 GPA versus females with a 3.5 GPA.

In [10]:
# Solution
odds.1 <- exp(-21.0680 + 6.1324*3.8 - 1.1697*1)
odds.2 <- exp(-21.0680 + 6.1324*3.5 - 1.1697*0)

odds.1 / odds.2

[1] 1.954276

Note that in this part, you are being asked to compare odds under conditions in which __two variables differ__: $\mathit{GPA}$ and $\mathit{SexM}$. So in this case, it isn't as simple as looking at $e^{\hat{\beta}_1}$ or $e^{\hat{\beta_2}}$.

### h.

Interpret the coefficient of $\mathit{GPA}$ in terms of an odds ratio.

In [11]:
# Solution
exp(6.1324)

[1] 460.5401

*Write your answer here. Double-click to edit.*

*Solution.* For fixed sex, the odds of acceptance are $e^{6.1324} = 460.540$ times higher with a one point increase in GPA.  

## Example 3

__Does the slope of GPA differ by sex?__

In other words, is the odds ratio of acceptance for a one-unit increase in GPA different for males versus females?

### a.

To allow the slope of GPA to differ by sex, we introduce an interaction term:

$$ \text{logit}(\pi) = \beta_0 + \beta_1 \mathit{GPA} + \beta_{2} \mathit{SexM} + \beta_3 (\mathit{GPA} \times \mathit{SexM}) $$

Fit this model.

In [12]:
# Solution
fit <- glm(Acceptance ~ GPA + Sex + GPA:Sex, data = MedGPA, family = binomial)
summary(fit)


Call:
glm(formula = Acceptance ~ GPA + Sex + GPA:Sex, family = binomial, 
    data = MedGPA)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-1.8936  -0.9547   0.2411   0.7240   2.3147  

Coefficients:
            Estimate Std. Error z value Pr(>|z|)  
(Intercept)  -24.385     10.082  -2.419   0.0156 *
GPA            7.083      2.891   2.450   0.0143 *
SexM           4.901     13.156   0.373   0.7095  
GPA:SexM      -1.709      3.709  -0.461   0.6450  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 75.791  on 54  degrees of freedom
Residual deviance: 53.729  on 51  degrees of freedom
AIC: 61.729

Number of Fisher Scoring iterations: 5


You should find the fitted model is

$$ \text{logit}(\hat{\pi}) = -24.385 + 7.083 \mathit{GPA} + 4.901 \mathit{SexM} - 1.709 (\mathit{GPA} \times \mathit{SexM}) $$

Therefore, for males, the model is
\begin{align*}
\text{logit}(\hat{\pi}) & = -24.385 + 7.083 \mathit{GPA} + 4.901 (1) - 1.709 (\mathit{GPA} \times (1))\\
& = -19.484 + 5.374 \mathit{GPA}
\end{align*}
and for females,
\begin{align*}
\text{logit}(\hat{\pi}) & = -24.385 + 7.083 \mathit{GPA} + 4.901 (0) - 1.709 (\mathit{GPA} \times (0))\\
& = -24.385 + 7.083 \mathit{GPA}
\end{align*}

### b.

Sketch the plots of the fitted model (in probability form) for males and females.
Use the axes provided at the end of the notes for Part 1.

*Solution.*  The vertical midpoint for the male model occurs at $\mathit{GPA} = 19.484 / 5.374 = 3.626$, and the vertical midpoint for the female model occurs at $\mathit{GPA} = 24.385 / 7.083 = 3.443$.

The slope at the vertical midpoint for the male model is $5.374 / 4 = 1.344$, and the slope at the vertical midpoint for the female model occurs at $7.083 /4 = 1.771$.

### c.

Estimate the odds ratio of acceptance for a unit increase in GPA for females.

In [13]:
# Solution
exp(7.083)

[1] 1191.538

### d.

Estimate the odds ratio of acceptance for unit increase in GPA for males.

In [14]:
# Solution
exp(5.374)

[1] 215.724

### e.

Answer the motivating question for this example. Justify your answer.

*Write your answer here. Double-click to edit.*

*Solution.* We fail to reject the null hypothesis of a $z$-test for the coefficient of $\mathit{GPA} \times \mathit{SexM}$, because the $p$-value = 0.6450 > 0.05 = $\alpha$. 

No, we do not see significant evidence that the slope of $\mathit{GPA}$ differs by sex.

### f.

Use the fitted model to estimate the probability of acceptance for a male with a 3.5 GPA.

In [15]:
# Solution
odds <- exp(-24.385 + 7.083*3.5 + 4.901*1 - 1.709*3.5*1)

odds / (1 + odds)

[1] 0.3373782

### g.

Should we drop both of the predictors that include sex?
Use a likelihood ratio test to compare the current model to the model with only GPA as a predictor.

In [16]:
# Solution
fit.GPA.only <- glm(Acceptance ~ GPA, data = MedGPA, family = binomial)
summary(fit.GPA.only)


Call:
glm(formula = Acceptance ~ GPA, family = binomial, data = MedGPA)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-1.7805  -0.8522   0.4407   0.7819   2.0967  

Coefficients:
            Estimate Std. Error z value Pr(>|z|)    
(Intercept)  -19.207      5.629  -3.412 0.000644 ***
GPA            5.454      1.579   3.454 0.000553 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 75.791  on 54  degrees of freedom
Residual deviance: 56.839  on 53  degrees of freedom
AIC: 60.839

Number of Fisher Scoring iterations: 4


In [17]:
# Solution
G = 56.839 - 53.729
p.value <- 1 - pchisq(G, 2)

G
p.value

[1] 3.11

[1] 0.2111894

*Write your answer here. Double-click to edit.*

*Solution.* 

Nested LRT:

1. $H_0: \beta_2 = \beta_3 = 0$ (reduced model) vs. $H_A: \text{at least one ($\beta_2$ or $\beta_3$) $\ne 0$}$ (full model)
2. G = 3.11
3. $p$-value = 0.2112
4. $p$-value > $\alpha$, so faill to reject $H_0$. We have significant evidence that the reduced model is sufficient. 